processor/src/main/java/org/mapstruct/ap/internal/conversion/DateToSqlTimestampConversion.java

```diff
@@ -0,0 +1,39 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.conversion;
+
+import org.mapstruct.ap.internal.model.common.ConversionContext;
+
+/**
+ * Conversion between {@link java.util.Date} and {@link java.sql.Timestamp}.
+ *
+ * @author Filip Hrisafov
+ */
+public class DateToSqlTimestampConversion extends SimpleConversion {
+
+    @Override
+    protected String getToExpression(ConversionContext conversionContext) {
+        return "new java.sql.Timestamp( <SOURCE>.getTime() )";
```

Comment:
I agree. However, I think we should do that only for `Time` and `Timestamp`. It is a bit dangerous doing it for `java.sql.Date` is a bit dangerous I think (it overlaps with `java.util.Date` and this one is pretty common).